pkg/tui/jira.go

```diff
@@ -0,0 +1,249 @@
+)
+
+type JiraModel struct {
+	width            int
+	height           int
+	columnSize       int
+	selectedIssue    int
+	focusedTab       int
+	maxPanels        int
+	showProjectsList bool
+	projectName      string
+	client           *jira.Client
+	projectItems     []jira.Project
+	projectsList     list.Model
+	issuesItems      []jira.Issue
+	issuesList       list.Model
+	projectKMap      projectKM
+}
+
+func NewJiraTui(client *jira.Client) JiraModel {
+
+	initProjectsList := list.New(make([]list.Item, 0), list.NewDefaultDelegate(), 0, 0)
+	initProjectsList.Title = "Projects"
+
+	initIssuesList := list.New(make([]list.Item, 0), list.NewDefaultDelegate(), 0, 0)
+	initIssuesList.Title = "Assigned Issues"
+
+	return JiraModel{
+		client:           client,
+		focusedTab:       1,
+		maxPanels:        3,
+		showProjectsList: false,
+		projectName:      "Selected Project Name",
+		projectsList:     initProjectsList,
+		issuesList:       initIssuesList,
+		projectKMap:      projectKMKeys,
+	}
+}
+
+func (m JiraModel) Init() tea.Cmd {
+	return tea.Batch(
+		getProjectsList(m.client),
+		getIssuesList(m.client),
+	)
+}
+
+func (m *JiraModel) NextPanel() {
+	m.focusedTab = (m.focusedTab + 1) % 4
+}
+
+func (m *JiraModel) PrevPanel() {
+	m.focusedTab = (m.focusedTab + 2) % 4
+}
+
+func (m *JiraModel) ToggleProjectSwitch() {
+	m.showProjectsList = !m.showProjectsList
+}
+
+func (m *JiraModel) setProjectListItems(projects []jira.Project) {
+	var items []list.Item
+	for _, project := range projects {
+		items = append(items, project)
+	}
+
+	m.projectsList = list.New(items, list.NewDefaultDelegate(), 30, 15)
+	m.projectsList.Title = "Projects"
+	m.projectsList.DisableQuitKeybindings()
+
+	m.projectItems = projects
+}
+
+func (m *JiraModel) setIssuesListItems(issues []jira.Issue) {
+	var items []list.Item
+	for _, project := range issues {
+		items = append(items, project)
+	}
+
+	m.issuesList = list.New(items, list.NewDefaultDelegate(), 30, 15)
+	m.issuesList.Title = "Assigned Issues"
+	m.issuesList.DisableQuitKeybindings()
+
+	m.issuesItems = issues
+}
+
+func (m JiraModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	var cmds []tea.Cmd
+	var cmd tea.Cmd
+
+	switch msg := msg.(type) {
+	case tea.KeyMsg:
+		switch msg.String() {
+		case "ctrl+c", "q":
+			return m, tea.Quit
+		case "tab":
+			m.NextPanel()
+		case "shift+tab":
+			m.PrevPanel()
+		case "S":
+			if m.focusedTab == 1 {
+				m.ToggleProjectSwitch()
+			}
+		}
+	case tea.WindowSizeMsg:
+		m.height = msg.Height
+		m.width = msg.Width
+		m.columnSize = msg.Width / 12
+	case projectsListResponse:
+		if msg.err != nil {
+			// TODO: Handle this better
+			log.Fatalf("Error with projects, err: %v", msg.err)
+		}
+		m.setProjectListItems(msg.projectsItems)
+	case issuesListResponse:
+		if msg.err != nil {
+			// TODO: Handle this better
+			log.Fatalf("Error with projects, err: %v", msg.err)
+		}
+		m.setIssuesListItems(msg.issuesItems)
+	}
+
+	if m.focusedTab == 1 {
```

Comment:
Can be constants since were checking for a specific tab (might not have merged this previously, if not theres a pr for it)